src/main/java/org/prebid/server/bidder/somoaudience/SomoaudienceBidder.java

```diff
@@ -40,98 +42,119 @@
 
+        return HttpRequest.of(HttpMethod.POST, url, body, headers, outgoingRequest);
+    }
+
+    private static ExtImpSomoaudience parseAndValidateImpExt(Imp imp) {
         final ObjectNode impExt = imp.getExt();
         if (impExt == null || impExt.size() == 0) {
-            throw new PreBidException(String.format("ignoring imp id=%s, extImpBidder is empty", impId));
+            throw new PreBidException(String.format("ignoring imp id=%s, extImpBidder is empty", imp.getId()));
         }
 
         final ExtImpSomoaudience extImpSomoaudience;
-
         try {
             extImpSomoaudience = Json.mapper.<ExtPrebid<?, ExtImpSomoaudience>>convertValue(imp.getExt(),
                     SOMOAUDIENCE_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
             throw new PreBidException(String.format(
                     "ignoring imp id=%s, error while decoding extImpBidder, err: %s", imp.getId(), e.getMessage()));
         }
-
-        return extImpSomoaudience.getPlacementHash();
+        return extImpSomoaudience;
     }
 
-    /**
-     * Creates {@link HttpRequest}s. One for each source id. Adds source id as url parameter
-     */
-    private Result<List<HttpRequest<BidRequest>>> createHttpRequests(Map<String, List<Imp>> placementsToImps,
-                                                                     List<BidderError> errors, BidRequest request) {
-        final List<HttpRequest<BidRequest>> httpRequests = new ArrayList<>();
-        for (Map.Entry<String, List<Imp>> placementIdToImps : placementsToImps.entrySet()) {
-            final String url = String.format("%s?s=%s", endpointUrl, placementIdToImps.getKey());
-            final BidRequest bidRequest = request.toBuilder().imp(placementIdToImps.getValue()).build();
-            final String bidRequestBody;
-            try {
-                bidRequestBody = Json.encode(bidRequest);
-            } catch (EncodeException e) {
-                errors.add(BidderError.badInput(String.format("error while encoding bidRequest, err: %s",
-                        e.getMessage())));
-                return Result.of(Collections.emptyList(), errors);
-            }
-            httpRequests.add(HttpRequest.of(HttpMethod.POST, url, bidRequestBody, headers, bidRequest));
-        }
-        return Result.of(httpRequests, errors);
+    private static MultiMap basicHeaders() {
+        return BidderUtil.headers()
+                .add("x-openrtb-version", "2.5");
     }
 
+    private static void addDeviceHeaders(MultiMap headers, BidRequest bidRequest) {
+        final Device device = bidRequest.getDevice();
+        headers.add("User-Agent", device.getUa());
+        headers.add("X-Forwarded-For", device.getIp());
+        headers.add("Accept-Language", device.getLanguage());
+
+        final Integer dnt = device.getDnt();
+        headers.add("DNT", dnt != null ? dnt.toString() : null);
```

Comment:
This is better but let's set header value to "0" if `device.dnt == null` in order to match behavior of Go version